examples/medplum-eligibility-demo/src/pages/CoveragePage.tsx

```diff
@@ -0,0 +1,107 @@
+import { Grid, Paper } from '@mantine/core';
+import { getReferenceString, Operator, resolveId, SearchRequest } from '@medplum/core';
+import { Coverage, Patient } from '@medplum/fhirtypes';
+import { Loading, PatientSummary, useMedplum, useMedplumNavigate } from '@medplum/react';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { CoverageActions } from '../components/actions/CoverageActions';
+import { CoverageDetails } from '../components/CoverageDetails';
+import { CoverageHeader } from '../components/CoverageHeader';
+
+export function CoveragePage(): JSX.Element {
+  const medplum = useMedplum();
+  const navigate = useMedplumNavigate();
+  const { id } = useParams() as { id: string };
+  const [coverage, setCoverage] = useState<Coverage | undefined>();
+  const [patient, setPatient] = useState<Patient>();
+
+  const tabs = [
+    ['Details', 'Details'],
+    ['History', 'History'],
+    ['Requests', 'Eligibility Requests'],
+    ['Responses', 'Eligibility Responses'],
+  ];
+
+  // Set the current tab to what is in the URL. If no tab, default to Details
+  const tab = window.location.pathname.split('/').pop();
+  const currentTab = tab && tabs.map((t) => t[0].toLowerCase()).includes(tab) ? tab : tabs[0][0].toLowerCase();
+
+  // Get a reference to the patient covered by the Coverage
+  const patientReference = coverage?.beneficiary;
+
+  useEffect(() => {
+    const fetchCoverage = async (): Promise<void> => {
+      try {
+        // Get the coverage details for the given resource. For more details on Coverage, see https://www.medplum.com/docs/billing/patient-insurance
+        const coverageData = await medplum.readResource('Coverage', id);
+        setCoverage(coverageData);
+      } catch (err) {
+        console.error(err);
+      }
+    };
+
+    const fetchLinkedPatient = async (): Promise<void> => {
+      if (patientReference) {
+        const patientId = resolveId(patientReference) as string;
+        try {
+          // Search for the details of the patient covered by the Coverage
+          const patientData = await medplum.readResource('Patient', patientId);
+          setPatient(patientData);
+        } catch (err) {
+          console.error(err);
```

Comment:
Throw the error here has well - this should get caught by the `ErrorBoundary` in APpShell